frontend/src/views/Main/components/Sidebar/Sidebar.tsx

```diff
@@ -1,32 +1,76 @@
+import { useNavigate } from "react-router-dom";
 import ChatListItem from "./components/ChatListItem/ChatListItem";
 import ComposeIcon from "./components/ComposeIcon/ComposeIcon";
 import SearchBar from "./components/SearchBar/SearchBar";
 import styles from "./Sidebar.module.css";
+import { socket } from "../../../../context/socket";
+import { useContext, useEffect, useState } from "react";
+import { socketContext } from "../../../../context/SocketContext";
 
 const Sidebar = () => {
+	const navigate = useNavigate();
+	const [rooms, setAllRooms] = useState<string[]>([]);
+	const { currentChat, setChat } = useContext(socketContext);
+	const [isCreating, setIsCreating] = useState(false);
+
+	const goHome = () => {
+		if (currentChat !== "") {
+			socket.emit("leave-chatroom", currentChat);
+
+			setChat("");
+
+			navigate("/");
+		}
+	};
+
+	const createChatroom = (e: any) => {
```

Comment:
Is it possible to type this with something specific?

Reply:
when typing it with React.FormEvent<HTMLFormElement>. It like doesn't understand the data tree correctly